src/halmos/cheatcodes.py

```diff
@@ -216,6 +219,58 @@ def symbolic_storage(ex, arg, sevm, stack, step_id):
     ex.storage[account_alias].symbolic = True
 
 
+def create_calldata_file_contract(ex, arg, sevm, stack, step_id):
+    filename = name_of(extract_string_argument(arg, 0))
+    contract_name = name_of(extract_string_argument(arg, 1))
+    return create_calldata_generic(
+        ex, arg, sevm, stack, step_id, contract_name, filename
+    )
+
+
+def create_calldata_contract(ex, arg, sevm, stack, step_id):
+    contract_name = name_of(extract_string_argument(arg, 0))
+    return create_calldata_generic(ex, arg, sevm, stack, step_id, contract_name)
+
+
+def create_calldata_generic(
+    ex, arg, sevm, stack, step_id, contract_name, filename=None
+):
+    """
+    Generate arbitrary symbolic calldata for the given contract.
+
+    Dynamic-array arguments are sized in the same way of regular test functions.
+    TODO: generate multiple calldata based on various size combinations.
+
+    The contract is identified by its contract name and optional filename.
+    TODO: provide variants that require only the contract address.
+    """
+    contract_json = BuildOut().get_by_name(contract_name, filename)[0]
+
+    abi = contract_json["abi"]
+    methodIdentifiers = contract_json["methodIdentifiers"]
+
+    results = []
+
+    for funsig in methodIdentifiers:
+        funname = funsig.split("(")[0]
+        funselector = methodIdentifiers[funsig]
+        funinfo = FunctionInfo(funname, funsig, funselector)
+
+        dyn_param_size = []
+
+        calldata = ByteVec()
+        calldata.append(int(funselector, 16).to_bytes(4, "big"))
+        mk_calldata(abi, funinfo, calldata, dyn_param_size, sevm.options)
```

Comment:
todo: there is an issue with calling mk_calldata multiple times, where the generated symbol names are not unique across different mk_calldata calls.

Reply:
fixed in https://github.com/a16z/halmos/pull/360/commits/ca1abbfba0e3f7643d62dc0d4a5e86e1fe1ddc81